### ISSUE TITLE
url parsing fails if front slash before fragment is missing and no path specified

### ISSUE THREAD

**Issue by holger-waschke** [go-openapi/strfmt#131]
we send azure alerts to alertmanager. alertmanager parse the url

https://github.com/prometheus/alertmanager/blob/d7ad5e12f8f823ec64e4e84e91d926b6a8e60f71/cli/alert_add.go#L127C25-L127C28

https://portal.azure.com#@930d042f-8145....

**Comment by fredbi** [comment]
That is correct: https://portal.azure.com#@930d042 (fragment following an empty path) is a valid URI.
We rely on go stdlib url.ParseRequestURI() to validate. And that one is confused by this construct:

https://portal.azure.com/#@930d042f-8145   -> with a slash following the hostname is less prone to errors.
